WISE-Community/WISE-Client
# feat(OpenResponse): Add FeedbackRuleEvaluator

## Changes
- Evaluate FeedbackRules and provide feedback if authored for the OpenResponse content
   - If the OR component content has cRater.feedback.enabled and cRater.feedback.rules is non-empty, OR will use the FeedbackRuleEvaluator to provide feedback instead of other feedback mechanisms (e.g. MultipleRoundsFeedback). 
   - The author can specify one or more feedback texts for a FeedbackRule.
- Refactor code to share FeedbackRule classes (evaluator, authoring view, CRater* classes, etc) between DialogGuidance and OR

## Test: Authoring
In the OR advanced component editing view:
- Verify that "Enable Feedback Rules" checkbox appears iff CRater is enabled
- Verify that "Scoring Rule" authoring section no longer appears when CRater is enabled
- Check "Enable Feedback Rules" checkbox
   - Verify that "isDefault" feedback is displayed as initial rule
   - Verify that the help button launches the Feedback Rule Authoring help dialog
   - Test that you can add, delete, reorder feedback rules as you can in DialogGuidance
      - Multiple feedback per rule is allowed (you can use the "+" to add more feedback per rule) 
- Un-check "Enable Feedback Rules" checkbox
   - Verify that FeedbackRules is not used 
   - If you enable the checkbox again, the rules that you authored before should appear again

DG authoring works as before

## Test: Student
- FeedbackRule evaluation for OR works and the correct feedback is shown to the student. 
   - If the rule has more than one feedback text, it should cycle through them.
- Existing C-Rater items that do not use FeedbackRule should work as before
- DG works as before

## Test: Grading
- Feedback text appears as auto-comment in the grade-by-step, grade-by-team, and revision views for the OR component

Closes #829
